web/src/attachments_ui.ts

```diff
@@ -223,3 +224,60 @@ export function set_up_attachments(): void {
         },
     });
 }
+
+export function suggest_delete_detached_attachments(attachments_list: ServerAttachment[]): void {
+    const html_body = render_confirm_delete_detached_attachments_modal({
+        attachments_list,
+        realm_allow_edit_history: realm.realm_allow_edit_history,
+    });
+
+    // Since we want to delete multiple attachments, we want to be
+    // able to keep track of attachments to delete and which ones to
+    // retry if it fails. A list is way too slow for the current use
+    // case. By Build a map from the list of attachments we want to
+    // delete, we can perform searches and deletion at O(1) time.
```

Comment:
Probably this comment should be be deleted; it's inaccurate that a list would be "way too slow". And I think the map data structure is fine without any extra explanation.